<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="18" passed="0" failed="17" skipped="1">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-11-11T16:33:57 IST" name="Suite" finished-at="2024-11-11T16:34:57 IST" duration-ms="60636">
    <groups>
      <group name="PL1">
        <method signature="RestAssured_TestCases_PL1.createAppointmentTest()[pri:1, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="createAppointmentTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.getAllApplicableDoctors()[pri:2, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="getAllApplicableDoctors" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.cancelAppointmentTest()[pri:3, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="cancelAppointmentTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.clashAppointmentTest()[pri:4, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="clashAppointmentTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.searchPatientTest()[pri:5, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="searchPatientTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.BookingListTest()[pri:6, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="BookingListTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.AllStockDetailsTest()[pri:7, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="AllStockDetailsTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.MainStoreTest()[pri:8, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="MainStoreTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.PharmacyStoreTest()[pri:9, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="PharmacyStoreTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.PatientConsumptionTest()[pri:10, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="PatientConsumptionTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.DeactivatePharmCountTest()[pri:12, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="DeactivatePharmCountTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.AppointApplicDeptTest()[pri:13, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="AppointApplicDeptTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.addcurrencyTest()[pri:15, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="addcurrencyTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.findMatchingPatientTest()[pri:16, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="findMatchingPatientTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.getAllRegisteredPatientsTest()[pri:17, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="getAllRegisteredPatientsTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.getAllBillingCounters()[pri:18, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="getAllBillingCounters" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.ActivatePharmCountTest()[pri:11, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="ActivatePharmCountTest" class="testcases.RestAssured_TestCases_PL1"/>
        <method signature="RestAssured_TestCases_PL1.AdmittedPatientsData()[pri:14, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" name="AdmittedPatientsData" class="testcases.RestAssured_TestCases_PL1"/>
      </group> <!-- PL1 -->
    </groups>
    <test started-at="2024-11-11T16:33:57 IST" name="Test" finished-at="2024-11-11T16:34:57 IST" duration-ms="60636">
      <class name="testcases.RestAssured_TestCases_PL1">
        <test-method signature="createAppointmentTest()[pri:1, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:33:57 IST" name="createAppointmentTest" description="Precondition: Create an appointment via the API
1. Send POST request to create a new appointment with provided data
2. Verify the response status code is 200 OK
3. Validate the response contains &#039;Status&#039; as &#039;OK&#039;
4. Retrieve and validate the Appointment ID from the response" finished-at="2024-11-11T16:33:57 IST" duration-ms="287" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "response" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "response" is null
at testcases.RestAssured_TestCases_PL1.createAppointmentTest(RestAssured_TestCases_PL1.java:51)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- createAppointmentTest -->
        <test-method signature="getAllApplicableDoctors()[pri:2, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:04 IST" name="getAllApplicableDoctors" description="Precondition: Multiple applicable doctors must exist
1. Validate that the response contains list of Doctors
2. Verify the response status code is 200.
3. Verify Performer IDs are unique" finished-at="2024-11-11T16:34:04 IST" duration-ms="8" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "getAllDoctorsResponse" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "getAllDoctorsResponse" is null
at testcases.RestAssured_TestCases_PL1.getAllApplicableDoctors(RestAssured_TestCases_PL1.java:80)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getAllApplicableDoctors -->
        <test-method signature="clashAppointmentTest()[pri:4, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:07 IST" name="clashAppointmentTest" description="Precondition: Patients and Doctor must be created successfully.
1. Send a GET request to fetch whether an appointment for the same time is created for the same doctor.
2. Verify the response status code is 200.
3. Validate the response indicates the status of clashes in appointment as true or false." finished-at="2024-11-11T16:34:07 IST" duration-ms="6" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.util.Map.get(Object)" because "clashedData" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.util.Map.get(Object)" because "clashedData" is null
at testcases.RestAssured_TestCases_PL1.clashAppointmentTest(RestAssured_TestCases_PL1.java:141)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- clashAppointmentTest -->
        <test-method signature="searchPatientTest()[pri:5, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:10 IST" name="searchPatientTest" description="Precondition: Patients and Doctor must be created successfully.
1. Send a GET request to fetch whether an appointment for the same time is created for the same doctor.
2. Verify the response status code is 200.
3. Validate the response indicates successful display of all the users that contain the string in their name." finished-at="2024-11-11T16:34:10 IST" duration-ms="4" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "searchedResponse" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "searchedResponse" is null
at testcases.RestAssured_TestCases_PL1.searchPatientTest(RestAssured_TestCases_PL1.java:173)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- searchPatientTest -->
        <test-method signature="BookingListTest()[pri:6, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:13 IST" name="BookingListTest" description="Precondition: Appointments must be made between current date and 5 days before the current date.
1. Send a GET request to fetch whether an appointment for the same time is created for the same doctor.
2. Verify the response status code is 200.
3. Validate the response indicates successfull display of appointments along with patient Id and Appointment time." finished-at="2024-11-11T16:34:13 IST" duration-ms="120" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.util.Map.get(Object)" because "searchResult" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.util.Map.get(Object)" because "searchResult" is null
at testcases.RestAssured_TestCases_PL1.BookingListTest(RestAssured_TestCases_PL1.java:217)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- BookingListTest -->
        <test-method signature="AllStockDetailsTest()[pri:7, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:17 IST" name="AllStockDetailsTest" description="Precondition: Appointments must be made between current date and 5 days before the current date.
1. Send a GET request to fetch whether an appointment for the same time is created for the same doctor.
2. Verify the response status code is 200.
3. Validate the response indicates successfull display of appointments along with patient Id and Appointment time." finished-at="2024-11-11T16:34:17 IST" duration-ms="6" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "stockDetails" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "stockDetails" is null
at testcases.RestAssured_TestCases_PL1.AllStockDetailsTest(RestAssured_TestCases_PL1.java:266)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- AllStockDetailsTest -->
        <test-method signature="MainStoreTest()[pri:8, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:20 IST" name="MainStoreTest" description="1. Send a GET request to fetch Main Store from the Pharmacy Settings.
2. Verify the response status code is 200.
3. Validate the response has an Id corresponding to the store along with the name and store description." finished-at="2024-11-11T16:34:20 IST" duration-ms="3" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "stockDetails" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "stockDetails" is null
at testcases.RestAssured_TestCases_PL1.MainStoreTest(RestAssured_TestCases_PL1.java:298)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- MainStoreTest -->
        <test-method signature="PharmacyStoreTest()[pri:9, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:23 IST" name="PharmacyStoreTest" description="Precondition: Some Pharmacy Stores must be created already. 
1. Send a GET request to fetch whether we are able to fetch the pharmacy stores or not.
2. Verify the response status code is 200.
3. Validate the response indicates successfull display of name of the store along with Store Id." finished-at="2024-11-11T16:34:23 IST" duration-ms="4" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "pharmacyStoreResponse" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "pharmacyStoreResponse" is null
at testcases.RestAssured_TestCases_PL1.PharmacyStoreTest(RestAssured_TestCases_PL1.java:334)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- PharmacyStoreTest -->
        <test-method signature="PatientConsumptionTest()[pri:10, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:26 IST" name="PatientConsumptionTest" description="1. Send a GET request to fetch whether we are able to fetch patient&#039;s consumption.
2. Verify the response status code is 200.
3. Validate the response indicates successfull display of patient name, patient id." finished-at="2024-11-11T16:34:26 IST" duration-ms="3" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "consumptionResponse" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "consumptionResponse" is null
at testcases.RestAssured_TestCases_PL1.PatientConsumptionTest(RestAssured_TestCases_PL1.java:370)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- PatientConsumptionTest -->
        <test-method signature="ActivatePharmCountTest()[pri:11, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:29 IST" name="ActivatePharmCountTest" description="Pre-conditions: Will require the counter Id and counterName to enter as a qeury paramter in the API. 
1. Send a PUT request to see whether we are able to activate the pharmacy counter.
2. Verify the response status code is 200.
3. Validate the response indicates successfull display of counter Id and counterName" finished-at="2024-11-11T16:34:29 IST" duration-ms="3" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.util.Map.get(Object)" because "searchResult" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.util.Map.get(Object)" because "searchResult" is null
at testcases.RestAssured_TestCases_PL1.ActivatePharmCountTest(RestAssured_TestCases_PL1.java:407)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- ActivatePharmCountTest -->
        <test-method signature="DeactivatePharmCountTest()[pri:12, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:32 IST" name="DeactivatePharmCountTest" description="1. Send a PUT request to fetch whether we are able to deactivate the pharmacy counter.
2. Verify the response status code is 200.
3. Validate the response indicates successfull display of status code as 200." finished-at="2024-11-11T16:34:32 IST" duration-ms="11" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "consumptionResponse" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "consumptionResponse" is null
at testcases.RestAssured_TestCases_PL1.DeactivatePharmCountTest(RestAssured_TestCases_PL1.java:448)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- DeactivatePharmCountTest -->
        <test-method signature="AppointApplicDeptTest()[pri:13, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:35 IST" name="AppointApplicDeptTest" description="1. Send a GET request to fetch a list of Appointment Applicable Departments.
2. Verify the response status code is 200.
3. Validate the response indicates successfull display of department name, department id, department code" finished-at="2024-11-11T16:34:35 IST" duration-ms="12" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "appointResponse" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "appointResponse" is null
at testcases.RestAssured_TestCases_PL1.AppointApplicDeptTest(RestAssured_TestCases_PL1.java:476)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- AppointApplicDeptTest -->
        <test-method signature="AdmittedPatientsData()[pri:14, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:38 IST" name="AdmittedPatientsData" description="1. Send a GET request to fetch a list of currently Admitted Patients Data.
2. Verify the response status code is 200.
3. Validate the response indicates successfull display of Patient Admission Id, Admitted Date but Discharged Date must be null" finished-at="2024-11-11T16:34:38 IST" duration-ms="1" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "admittedPatientResponse" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "admittedPatientResponse" is null
at testcases.RestAssured_TestCases_PL1.AdmittedPatientsData(RestAssured_TestCases_PL1.java:511)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- AdmittedPatientsData -->
        <test-method signature="addcurrencyTest()[pri:15, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:42 IST" name="addcurrencyTest" description="Precondition: Currency code must not already exist
1. Validate the currency code
2. Validate created by
3. Validate created on date
4. Validate the status code as 200." finished-at="2024-11-11T16:34:42 IST" duration-ms="59" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.util.Map.put(Object, Object)" because "postData" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.util.Map.put(Object, Object)" because "postData" is null
at testcases.RestAssured_TestCases_PL1.addcurrencyTest(RestAssured_TestCases_PL1.java:559)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- addcurrencyTest -->
        <test-method signature="findMatchingPatientTest()[pri:16, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:45 IST" name="findMatchingPatientTest" description="Precondition: Matching Patient must exist
1. Validate that the appointment ID is not null.
2. Send a GET request to find patient that has same phone number.
3. Verify the response status code is 200." finished-at="2024-11-11T16:34:45 IST" duration-ms="2" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.util.Map.get(Object)" because "postData" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.util.Map.get(Object)" because "postData" is null
at testcases.RestAssured_TestCases_PL1.findMatchingPatientTest(RestAssured_TestCases_PL1.java:618)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- findMatchingPatientTest -->
        <test-method signature="getAllRegisteredPatientsTest()[pri:17, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:48 IST" name="getAllRegisteredPatientsTest" description="1. Send a GET request to get all the registered patients
2. Validate that all the patient IDs are unique.
3. Verify the response status code is 200." finished-at="2024-11-11T16:34:48 IST" duration-ms="32" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "matchingPatientResponse" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "matchingPatientResponse" is null
at testcases.RestAssured_TestCases_PL1.getAllRegisteredPatientsTest(RestAssured_TestCases_PL1.java:665)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getAllRegisteredPatientsTest -->
        <test-method signature="getAllBillingCounters()[pri:18, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:51 IST" name="getAllBillingCounters" description="1. Send a GET request to get all billing counters
2. Validate that all the counter IDs are unique.
3. Verify the response status code is 200." finished-at="2024-11-11T16:34:51 IST" duration-ms="26" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "io.restassured.response.Response.statusCode()" because "billingCounterResponse" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "io.restassured.response.Response.statusCode()" because "billingCounterResponse" is null
at testcases.RestAssured_TestCases_PL1.getAllBillingCounters(RestAssured_TestCases_PL1.java:697)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getAllBillingCounters -->
        <test-method signature="cancelAppointmentTest()[pri:3, instance:testcases.RestAssured_TestCases_PL1@44e3a2b2]" started-at="2024-11-11T16:34:54 IST" name="cancelAppointmentTest" description="Precondition: An appointment must be created successfully.
1. Validate that the appointment ID is not null.
2. Send a PUT request to cancel the appointment using the appointment ID.
3. Verify the response status code is 200.
4. Validate the response indicates successful cancellation." finished-at="2024-11-11T16:34:54 IST" depends-on-methods="testcases.RestAssured_TestCases_PL1.createAppointmentTest" duration-ms="0" status="SKIP">
          <exception class="java.lang.Throwable">
            <message>
              <![CDATA[Method testcases.RestAssured_TestCases_PL1.cancelAppointmentTest() on instance testcases.RestAssured_TestCases_PL1@44e3a2b2 depends on not successfully finished methods 
[testcases.RestAssured_TestCases_PL1.createAppointmentTest() on instance testcases.RestAssured_TestCases_PL1@44e3a2b2]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.Throwable: Method testcases.RestAssured_TestCases_PL1.cancelAppointmentTest() on instance testcases.RestAssured_TestCases_PL1@44e3a2b2 depends on not successfully finished methods 
[testcases.RestAssured_TestCases_PL1.createAppointmentTest() on instance testcases.RestAssured_TestCases_PL1@44e3a2b2]
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:156)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.Throwable -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- cancelAppointmentTest -->
      </class> <!-- testcases.RestAssured_TestCases_PL1 -->
    </test> <!-- Test -->
  </suite> <!-- Suite -->
</testng-results>
